component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java

```diff
@@ -311,7 +307,7 @@ private boolean validCertExistsAsAttribute(HttpServletRequest request) {
      * @param requestCert    X.509 certificate presented to server during TLS hand shake.
      * @return Whether the client was successfully authenticated or not.
      */
-    protected boolean authenticate(X509Certificate registeredCert, X509Certificate requestCert)
+    protected boolean authenticate(X509Certificate registeredCert, X509Certificate requestCert, OAuthAppDO oAuthAppDO)
```

Comment:
Since this is a protected method let's introduce a new method and deprecate this one. also, remove all usages of this method.

Reply:
fixed